hpx/runtime/serialization/variant.hpp

```diff
@@ -0,0 +1,78 @@
+//  Copyright (c) 2015 Anton Bikineev
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_SERIALIZATION_VARIANT_HPP
+#define HPX_SERIALIZATION_VARIANT_HPP
+
+#include <hpx/runtime/serialization/serialize.hpp>
+
+#include <boost/variant.hpp>
+
+namespace hpx { namespace serialization
+{
+    namespace detail
+    {
+        struct variant_save_visitor: boost::static_visitor<>
+        {
+            variant_save_visitor(output_archive& ar)
+              : ar(ar)
+            {}
+
+            template <class T>
+            void operator()(const T& value)
+            {
+                ar << value;
+            }
+
+        private:
+            output_archive& ar;
+        };
+
+        template <class... Args>
+        struct load_helper_t{};
+
+        template <class Variant, class... Tail>
+        BOOST_FORCEINLINE void load_helper(input_archive& ar, Variant& var,
+                int which, load_helper_t<boost::detail::variant::void_, Tail...>)
+        {
+            HPX_ASSERT(false);
+        }
+
+        template <class Variant, class Head, class... Tail>
+        BOOST_FORCEINLINE void load_helper(input_archive& ar, Variant& var,
+                int which, load_helper_t<Head, Tail...>)
+        {
+            if (which == 0)
+            {
+                Head value; // default ctor concept
+                ar >> value;
+                var = value;
+            }
+            else
+            {
+                load_helper(ar, var, which - 1, load_helper_t<Tail...>());
+            }
+        }
+    }
+
+    template <class... Args>
+    void serialize(output_archive& ar, boost::variant<Args...>& variant, unsigned)
```

Comment:
That's a problem.
In case it's const ref partial ordering will take other overload defined in serialize.hpp:
https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/serialization/serialize.hpp#L17
On the other hand I guess we can change that one by splitting it into two serialize methods for output and input archive.